### PR TITLE
feat(release): add darkmode styling to framework

### PR DESCRIPTION
**Pull request by mvmagnussen**
Added dark mode styling to the design system framework.

Including a switch for toggling between light and dark mode in the header. The switch toggles automatically when the OS color preference is changed. 

Note: it is not possibly to toggle to light mode when dark mode is selected as OS color preference.

image

### CODE REVIEW

**Comment by kwltrs** [comment]
The toggle button looks strange on my machine (standard issued red hat and firefox).
image

**Comment by kwltrs** [comment]
The contrast seems a bit low on the import instructions and props sections.
image

**Comment by palok** [comment]
> The toggle button looks strange on my machine (standard issued red hat and firefox).
> image

Var dette med `npm start`? Og hvilken browser?

**Comment by ivarni** [comment]
The toggle looks fine here, Chrome 78 on Ubuntu

But I think the dropdowns are missing something
image

**Comment by mvmagnussen** [comment]
> The toggle looks fine here, Chrome 78 on Ubuntu
> 
> But I think the dropdowns are missing something
> image

This dropdown is a component issue (good catch), however, not relevant in this PR. See PR #819.

**Comment by mvmagnussen** [comment]
> The contrast seems a bit low on the import instructions and props sections.
> image

Fikset!

**Comment by mvmagnussen** [comment]
> The toggle button looks strange on my machine (standard issued red hat and firefox).
> image

Delete content in `dist` folder and then `npm run build:all` and/or `npm run styleguidist:build`, finally `npm start`. Then this should not be an issue.

**Comment by kwltrs** [comment]
> Delete content in `dist` folder and then `npm run build:all` and/or `npm run styleguidist:build`, finally `npm start`. Then this should not be an issue.

This did not fix the toggle. :(

File: src/styles/components/top-menu.less [hunk 1]
```diff
@@ -10,6 +10,22 @@
         margin: 10px 0;
         vertical-align: middle;
 
+        .native & {
```
Comment (antidecaf) [inline src/styles/components/top-menu.less:19]: Looks like this override has been inserted twice

**Comment by antidecaf** [comment]
Styling assigned to id's and element selectors, which is done a lot in this PR, isn't really a pattern that is used a lot elsewhere in the design system. Is it possible to in stead use (BEM-based) classes, for the sake of consistency? :)

**Comment by palok** [comment]
> Styling assigned to id's and element selectors, which is done a lot in this PR, isn't really a pattern that is used a lot elsewhere in the design system. Is it possible to in stead use (BEM-based) classes, for the sake of consistency? :)

New to BEM, but did these changes: 57cd62f1fbc4085e9ce19e0bed241b43b6c83964

**Review by kwltrs — APPROVED** [review]

**Pull request merged**